test/dynamo/test_subclasses.py

```diff
@@ -1104,6 +1105,38 @@ def backend(gm, args):
             )
             out = compile_fn(nt_view)
 
+    def test_zeros(self):
+        dtypes = (torch.float, torch.double, torch.half)
+        devices = ("cpu", "cuda")
+        for dtype, device in itertools.product(dtypes, devices):
+            if device == "cuda" and not TEST_CUDA:
+                continue
+
+            kwargs = {
+                "device": device,
+                "dtype": dtype,
+            }
+            # Need more extensive testing with various settings dtype/device etc.
+            x, _ = self._get_jagged_tensor(((2, 3, 4), 3), None, requires_grad=True)
+
+            def fn1(nt):
+                out = torch.zeros(nt.shape, **kwargs)
+                return out
+
+            def fn2(nt):
+                out = torch.zeros(nt.shape[:2] + (2, 3), **kwargs)
+                return out
+
+            def do_check(fn):
+                compile_fn = torch.compile(
+                    fn, fullgraph=True, backend="aot_eager", dynamic=True
+                )
+                out = compile_fn(x)
+                self.assertEqual(out, fn(x))
+
+            do_check(fn1)
+            do_check(fn2)
```

Comment:
Parametrization over the test case does not work for dynamo's Test class unfortunately, so I updated to do it manually here. I didn't look too deeply, but no other dynamo test uses parametrization so I imagine it is not trivial to solve.